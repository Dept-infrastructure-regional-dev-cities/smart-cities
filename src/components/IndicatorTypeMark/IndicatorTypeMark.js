import React from 'react';
import PropTypes from 'prop-types';
import getColorVariant from '../../helpers/getColorVariant';
import style from './IndicatorTypeMark.scss';
import COLORS from '../../style/_colors.scss';

const classnames = require('classnames/bind').bind(style);

const SIZES = {
  SMALL: 'SMALL',
  LARGE: 'LARGE',
};

// A "P" in a circle
const IndicatorTypeMark = (props) => {
  const performanceMarkColor = getColorVariant(props.colorName, '900');

  let pColor;
  if (props.isCategoryPage) {
    pColor = COLORS.WHITE;
  } else if (props.size === SIZES.LARGE) {
    pColor = performanceMarkColor;
  } else {
    pColor = COLORS.GREY_700;
  }

  // Use a solid colour on the category pages
  // and just an outline on the overview page
  const pStyle = props.isCategoryPage
    ? {
      background: performanceMarkColor,
      color: pColor,
    }
    : {
      border: `1px solid ${performanceMarkColor}`,
      color: pColor,
    };

  const className = classnames(
    style.performanceMark,
    {
      performanceMarkSmall: props.size === SIZES.SMALL,
      performanceMarkLarge: props.size === SIZES.LARGE,
    },
    props.className,
  );

  return <div className={className} style={pStyle}>P</div>;
};

IndicatorTypeMark.propTypes = {
  className: PropTypes.string,
  colorName: PropTypes.string.isRequired,
  isCategoryPage: PropTypes.bool,
  size: PropTypes.oneOf(Object.keys(SIZES)).isRequired,
};

export default IndicatorTypeMark;
