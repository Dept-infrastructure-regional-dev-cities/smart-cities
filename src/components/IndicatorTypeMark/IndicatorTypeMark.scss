@import '~style/variables';

// note that color, background and border are set in JS
.performanceMark {
  border-radius: 100%;
  text-align: center;
  font-weight: 700;
  border: 1px solid transparent;
}

.performanceMarkSmall {
  width: 14px;
  height: 14px;
  line-height: 13px;
  font-size: 10px;
  text-indent: 1px;
}

.performanceMarkLarge {
  width: 22px;
  height: 22px;
  line-height: 22px;
  font-size: 13px;
}
