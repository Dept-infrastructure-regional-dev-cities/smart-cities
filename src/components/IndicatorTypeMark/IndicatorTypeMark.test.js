import React from 'react';
import { shallow } from 'enzyme';
import IndicatorTypeMark from './IndicatorTypeMark';

const defaultProps = {
  colorName: 'jobs',
  size: 'LARGE',
};

it('should match Snapshot', () => {
  const component = shallow(
    <IndicatorTypeMark {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});

it('should show a filled background and white P on category pages', () => {
  const component = shallow(
    <IndicatorTypeMark
      {...defaultProps}
      isCategoryPage
    />
  );

  const style = component.find('div').prop('style');
  expect(style.background).toBe('JOBS_900');
  expect(style.border).toBeUndefined();
  expect(style.color).toBe('WHITE');
});

it('should show a stroke and grey P on overview page when small', () => {
  const component = shallow(
    <IndicatorTypeMark
      {...defaultProps}
      size="SMALL"
    />
  );

  const style = component.find('div').prop('style');
  expect(style.background).toBeUndefined();
  expect(style.border).toBe('1px solid JOBS_900');
  expect(style.color).toBe('GREY_700');
});

it('should show a stroke and colored P on overview page when small', () => {
  const component = shallow(
    <IndicatorTypeMark
      {...defaultProps}
      size="LARGE"
    />
  );

  const style = component.find('div').prop('style');
  expect(style.background).toBeUndefined();
  expect(style.border).toBe('1px solid JOBS_900');
  expect(style.color).toBe('JOBS_900');
});
