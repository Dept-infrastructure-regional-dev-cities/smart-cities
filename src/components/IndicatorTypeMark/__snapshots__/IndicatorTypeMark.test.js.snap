// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div className=\\"performanceMark performanceMarkLarge\\" style={{...}}>
  P
</div>"
`;
